tests/mock_server.py

```diff
@@ -1,27 +1,110 @@
+import asyncio
+import aiohttp.web
 import pytest
 import requests
 import shutil
 import signal
 import subprocess as sp
 import sys
 import time
-
+import threading
 
 _proxy_bypass = {
   "http": None,
   "https": None,
 }
 
 
+class AIOServer(threading.Thread):
+    def __init__(self, port):
+        super().__init__(target=self._run)
+        self._loop = None
+        self._port = port
+        self.start()
+        self.endpoint_url = 'http://localhost:{}'.format(port)
+        self._shutdown_evt = asyncio.Event()
+
+    def _run(self):
+        self._loop = asyncio.new_event_loop()
+        app = aiohttp.web.Application(loop=self._loop)
+        app.router.add_route('*', '/ok', self.ok)
+        app.router.add_route('*', '/{anything:.*}', self.stream_handler)
+
+        try:
+            aiohttp.web.run_app(app, host='localhost', port=self._port,
```

Comment:
you can run aiohttp server in same thread without using threading. See example here https://github.com/aio-libs/aiozipkin/blob/5f914c6eb811db046a58d5d816cd19223ae2435c/demos/queue/runner.py#L7-L25 it is a bit broken I wrote those late night, but you get the idea

Reply:
cool, want me migrate?

Reply:
yes please.

Reply:
i just realized I shutdown the server via `loop.stop` however we cannot do this in a unittest so I think we need to keep it in a thread.